### PR TITLE
Clarify indentation of arguments of piped functions

### LINKED ISSUE #22834: F# code formatting guidelines: Clarify indentation of arguments of piped functions
The F# code formatting guidelines, section [Formatting function parameter application](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#formatting-function-parameter-application), contains the following example:

```f#
list1
|> List.iter
    (fun elem ->
        printfn $"%d{a + elem}")
```

Above, the parameter (the `(` to be specific) is indented 4 spaces relative to the start of the `|>` operator:

image

However, consider how this looks for 2 spaces of indentation:

```f#
list1
|> List.iter
  (fun elem ->
    printfn $"%d{a + elem}")
```

IMHO this doesn't look great; the argument to `List.iter` looks deindented relative to `List.iter`.

Compare this with the style guide example given just above this one, with no pipeline:

```f#
List.iter
    (fun elem ->
        printfn $"%d{a + elem}")
    list1
```

This, of course, looks good with 2 spaces of indentation:

```f#
List.iter
  (fun elem ->
    printfn $"%d{a + elem}")
  list1
```

I propose that we specify how parameters to pipelined functions are indented, in a way that ensures it looks good also on 2 spaces (and possibly others, though the style guide only recommends 4 or 2 spaces, so think that should be the focus).

For example:

> For function application in a pipeline, the arguments should be indented relative to the function:
> ```f#
> // With 4 spaces indentation
> list1
> |> List.iter
>        someLongParam
>        anotherLongParam
>
> // With 2 spaces indentation
> list1
> |> List.iter
>      someLongParam
>      anotherLongParam
> ```

Or alternatively, closer to today's format in the 4-space case:

> For function application in a pipeline, the arguments should be indented 2 spaces relative to the function:
> ```f#
> list1
> |> List.iter
>      someLongParam
>      anotherLongParam
> ```

Or alternatively, preserving today's formatting in the 4-space case:

> For function application in a pipeline, the arguments should be indented relative to the pipe symbol, but always at least 1 space from the function:
> ```f#
> // With both 4 and 2 spaces indentation
> list1
> |> List.iter
>     someLongParam
>     anotherLongParam
> ```

I'm open to other suggestions. I can also create a PR once it's decided.

### PR DESCRIPTION
**Pull request by cmeeren**
Fixes #22834

### CODE REVIEW

**Comment by cmeeren** [comment]
Actually, this one needs one more change for consistency. I'll let you know when I've pushed it.

**Comment by cmeeren** [comment]
Ready for review.

The additional change I mentioned was pushed in [this commit](https://github.com/dotnet/docs/pull/23093/commits/b9909d08918c5d58d927771ca6c901e34182080b). The style guide was not consistent on whether indentation of a lambda (`fun ...` was relative to `(` or `fun`. Since the first commit in this PR essentially required indentation relative to `fun` in order for the style guide to be consistent, I went ahead and clarified that and fixed the samples.

File: docs/fsharp/style-guide/formatting.md [hunk 1]
```diff
@@ -900,18 +900,44 @@ For lambda expressions, you may also want to consider placing the body of a lamb
...
     |> List.iter
-        (fun elem ->
-            printfn $"%d{a + elem}")
+           (fun elem ->
+                printfn $"A very long line to format the value: %d{a + elem}")
 ```
 
 If the body of a lambda expression is multiple lines long, you should consider refactoring it into a locally-scoped function.
 
+The parameters should always be indented relative to the function or `fun`/`function` keyword, regardless of the context in which the function appears:
```
Comment (cartermp) [inline docs/fsharp/style-guide/formatting.md:915]:
```suggestion
The parameters should generally be indented relative to the function or `fun`/`function` keyword, regardless of the context in which the function appears:
```

Comment (cartermp) [reply docs/fsharp/style-guide/formatting.md:915]: Let's try to be a little softer on the language here.

Comment (cmeeren) [reply docs/fsharp/style-guide/formatting.md]: Done!

**Review by cartermp — APPROVED** [review]
Approved pending CI passing, thanks!

**Pull request merged**